<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>GPIO</class>
 <widget class="QWidget" name="GPIO">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>743</width>
    <height>300</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QTableView" name="tableView">
     <property name="selectionBehavior">
      <enum>QAbstractItemView::SelectRows</enum>
     </property>
     <property name="sortingEnabled">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QWidget" name="widget" native="true">
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>Port</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="cPort"/>
      </item>
      <item>
       <widget class="QLabel" name="label">
        <property name="text">
         <string>from</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="sFrom">
        <property name="maximum">
         <number>15</number>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>to</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="sTo">
        <property name="maximum">
         <number>15</number>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>dir</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="cMode"/>
      </item>
      <item>
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>pull</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="cPull"/>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QCheckBox" name="cLvl">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>state</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="bAdd">
        <property name="text">
         <string>ADD</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="bDel">
        <property name="text">
         <string>DEL</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="bRead">
        <property name="text">
         <string>READ</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>bAdd</sender>
   <signal>clicked()</signal>
   <receiver>GPIO</receiver>
   <slot>bAdd()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>526</x>
     <y>270</y>
    </hint>
    <hint type="destinationlabel">
     <x>298</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cPort</sender>
   <signal>currentIndexChanged(QString)</signal>
   <receiver>GPIO</receiver>
   <slot>port_changed(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>85</x>
     <y>270</y>
    </hint>
    <hint type="destinationlabel">
     <x>298</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sFrom</sender>
   <signal>valueChanged(int)</signal>
   <receiver>sTo</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>186</x>
     <y>270</y>
    </hint>
    <hint type="destinationlabel">
     <x>256</x>
     <y>270</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tableView</sender>
   <signal>clicked(QModelIndex)</signal>
   <receiver>GPIO</receiver>
   <slot>onTableClicked(QModelIndex)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>349</x>
     <y>126</y>
    </hint>
    <hint type="destinationlabel">
     <x>349</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cMode</sender>
   <signal>currentIndexChanged(QString)</signal>
   <receiver>GPIO</receiver>
   <slot>dir_changed(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>342</x>
     <y>270</y>
    </hint>
    <hint type="destinationlabel">
     <x>371</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>bAdd()</slot>
  <slot>port_changed(QString)</slot>
  <slot>onTableClicked(QModelIndex)</slot>
  <slot>dir_changed(QString)</slot>
 </slots>
</ui>
